whisper.py

```diff
@@ -920,13 +928,21 @@ def file_fetch(fh, fromTime, untilTime, now=None):
     untilTime = now
 
   diff = now - fromTime
+
   for archive in header['archives']:
-    if archive['retention'] >= diff:
+    if not archiveToSelect and archive['retention'] >= diff:
       break
+    if archiveToSelect and archive['secondsPerPoint'] == archiveToSelect:
+      break
+    elif archiveToSelect:
+      archive = None
```

Comment:
This might be easier to read as a nested if:
```
if archiveToSelect:
  if archive['secondsPerPoint'] == archiveToSelect:
    break
  archive = None
else:
  if archive['retention'] >= diff:
    break
```